src/check_executor.rs

```diff
@@ -170,11 +205,27 @@ async fn executor_loop(
                         job_checker.check_url(config, tick, &job_region).await
                     };
 
+                    let will_retry = check_result.status == CheckStatus::Failure
+                        && scheduled_check.retry_count < conf.failure_retries;
+
+                    record_result_metrics(
+                        &check_result,
+                        scheduled_check.retry_count > 0,
+                        will_retry,
+                    );
+
+                    // re-queue for execution again
+                    if will_retry {
+                        tracing::debug!(result = ?check_result, "executor.check_will_retry");
+                        job_check_sender.queue_check_for_retry(scheduled_check);
+                        job_num_running.fetch_sub(1, Ordering::SeqCst);
+                        return;
+                    }
+
                     if let Err(e) = job_producer.produce_checker_result(&check_result) {
                         tracing::error!(error = ?e, "executor.failed_to_produce");
                     }
 
-                    record_result_metrics(&check_result);
                     tracing::debug!(result = ?check_result, "executor.check_complete");
 
                     scheduled_check.record_result(check_result);
```

Comment:
Is there any problem with delaying the recording of the result here? I forget if anything is waiting on it